icu4c/source/i18n/zonemeta.cpp

```diff
@@ -290,9 +290,13 @@ ZoneMeta::getCanonicalCLDRID(const UnicodeString &tzid, UErrorCode& status) {
     UResourceBundle *rb = ures_getByKey(top, gTypeMapTag, NULL, &tmpStatus);
     ures_getByKey(rb, gTimezoneTag, rb, &tmpStatus);
     ures_getByKey(rb, id, rb, &tmpStatus);
-    if (U_SUCCESS(tmpStatus)) {
+    int32_t bcp47StrLen = 0;
+    ures_getString(rb, &bcp47StrLen, &tmpStatus);
+    if (U_SUCCESS(tmpStatus) && bcp47StrLen!=4) {
         // type entry (canonical) found
         // the input is the canonical ID. resolve to const UChar*
+        // TODO: For now we skip entries mapping to metazone short ids,
+        // which have 4 characters.
```

Comment:
TODO should link to a bug

Reply:
Yeah, that is https://unicode-org.atlassian.net/browse/CLDR-15023 which I had not filed at the time. I will add the reference during the next CLDR integration (for alpha3 or beta1)